src/content/app/tools/blast/components/blast-genomic-hits-diagram/BlastGenomicHitsDiagram.tsx

```diff
@@ -0,0 +1,133 @@
+
+import styles from './BlastGenomicHitsDiagram.scss';
+
+type Props = {
+  genomeId: string;
+  job: BlastJobResult;
+  width: number;
+};
+
+const BlastGenomicHitsDiagram = (props: Props) => {
+  const { genomeId, job, width } = props;
+  const [isExpanded, setIsExpanded] = useState(false);
+  const { currentData: genomeKaryotype } = useGenomeKaryotypeQuery(genomeId); // the karyotype data has properly sorted top-level sequences
+
+  if (!genomeKaryotype) {
+    return null;
+  }
+
+  const topMatches = getTopMatchesWithRegionNames(job);
+
+  // For the compact view, select regions with top five matches, and deduplicate them
+  const regionNamesForCompactView = new Set(
+    topMatches.map(({ regionName }) => regionName)
+  );
+  const sortedRegionNamesForCompactView = genomeKaryotype
+    .map(({ name }) => name)
+    .filter((name) => regionNamesForCompactView.has(name));
+
+  const regionNamesForExpandedView = new Set(
+    job.hits.map((hit) => hit.hit_acc)
+  );
+  const sortedRegionNamesForExpandedView = genomeKaryotype
+    .map(({ name }) => name)
+    .filter((name) => regionNamesForExpandedView.has(name));
+
+  const remainderToExpandedRegionNamesList =
+    sortedRegionNamesForExpandedView.length -
+    sortedRegionNamesForCompactView.length;
+
+  const regionNamesForDisplay = isExpanded
+    ? sortedRegionNamesForExpandedView
+    : sortedRegionNamesForCompactView;
+
+  return (
+    <div>
+      {regionNamesForDisplay.map((name) => (
+        <BlastGenomicRegionHeatmap
+          key={name}
+          job={job}
+          width={width}
+          regionName={name}
+          topHits={topMatches.filter((match) => match.regionName === name)}
+        />
+      ))}
+      <RegionsExpandToggle
+        count={remainderToExpandedRegionNamesList}
+        isExpanded={isExpanded}
+        onClick={() => setIsExpanded(!isExpanded)}
+      />
+    </div>
+  );
+};
+
+const RegionsExpandToggle = (props: {
+  count: number;
+  onClick: () => void;
+  isExpanded: boolean;
+}) => {
+  const { count, onClick, isExpanded } = props;
+  const isCollapsed = !isExpanded;
+
+  if (!count) {
+    return null;
+  }
+
+  return (
+    <div className={styles.regionsExpandToggle}>
+      {isCollapsed ? (
+        <>
+          <span>Other regions with hits</span>
+          <PillButton onClick={onClick}>+{count}</PillButton>
+        </>
+      ) : (
+        <CloseButton onClick={onClick} />
+      )}
+    </div>
+  );
+};
+
+const getTopMatchesWithRegionNames = (job: BlastJobResult) => {
+  // order all hit_hsps by their e-value, and return regions containing the top five
+
+  const hitHspsWithRegionName = job.hits.flatMap((hit) => {
+    return hit.hit_hsps.map((hit_hsp) => ({
+      start: Math.min(hit_hsp.hsp_hit_from, hit_hsp.hsp_hit_to),
+      end: Math.max(hit_hsp.hsp_hit_from, hit_hsp.hsp_hit_to),
+      eValue: hit_hsp.hsp_expect,
+      regionName: hit.hit_acc
+    }));
+  });
+
+  const sortedHsps = sortBy(hitHspsWithRegionName, 'eValue');
```

Comment:
The order of top hits in svg and table doesn't match.

Reply:
The Patches are excluded from the image, but not the table, yes ?